cupy/core/carray.pxi

```diff
@@ -122,9 +123,21 @@ cpdef str _get_header_source():
     return _header_source
 
 
+cdef inline str _convert_cuComplex_to_Thrust(str source):
+    source = re.sub(r'<cuComplex.h>',
```

Comment:
Ah, what I meant by "regex that matches the whole line" was like this:

```py
# module level (to precompile the regex)
cucomplex_include_tokens = ['', '#', 'include', '<', r'cuComplex\.h', '>', '']
cucomplex_include_pattern = re.compile(r'\s*'.join(cucomplex_include_tokens))

# in _convert_cuComplex_to_Thrust
lines = []
for line in source.splitlines(keepends=True):
    if cucomplex_include_pattern.match(line):
        lines += '#include <cupy/cuComplex_bridge.h>  // translate_cucomplex'
    else:
        lines += line
return ''.join(lines)
```

Reply:
I see. Sorry I know nothing about regex. While your regex code looks fine to me, for some reason it can't detect the following cases:
```python
>>> cucomplex_include_pattern.match(r'#include <cuComplex.h>')  # ok
<_sre.SRE_Match object; span=(0, 22), match='#include <cuComplex.h>'>
>>> cucomplex_include_pattern.match(r'//#include <cuComplex.h>')  # miss!
>>> cucomplex_include_pattern.match(r'// #include <cuComplex.h>')  # miss!
>>> cucomplex_include_pattern.match(r'// # include <cuComplex.h> ')  #miss!
```
@kmaehashi could you help me diagnose please?

Reply:
```py
>>> cucomplex_include_pattern.match(r'//#include <cuComplex.h>')  # miss!
>>> cucomplex_include_pattern.match(r'// #include <cuComplex.h>')  # miss!
>>> cucomplex_include_pattern.match(r'// # include <cuComplex.h> ')  #miss!
```

This behavior is what I intended. They are commented out so we don't have to replace it.

Reply:
Oh oh, I see! I misunderstood what you meant. Will make a change before going to bed :)

Reply:
@kmaehashi This is added now.